python/cuda_parallel/tests/test_reduce.py

```diff
@@ -2,10 +2,19 @@
 #
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+# TODO(rwgk): Format entire file.
+# fmt: off
+
+import cupy as cp
 import numpy
 import pytest
-from numba import cuda
+import random
+import re
+import numba.cuda
+import numba.types
 import cuda.parallel.experimental as cudax
+from cuda.parallel.experimental import _iterators
```

Comment:
If we need something from a non-public submodule in the tests, then it's possible that one of 

- it should go in a public API
- we don't really need it

For instance, we are using `_iterators.pointer()` to construct inputs for one of our tests. This suggests that `pointer()` should be a public API (OR we are testing something that we don't expect users to ever do).

Reply:
I think it might be a leftover. `_iterators.pointer()` is an implementation detail for transform iterator (glue layer to make it support containers). I would suggest to avoid testing reduce with pointer directly and test only reduction of transformed `cp.array`.

Reply:
Summary of short offline discussion: Maybe in a follow-on PR:
```
TransformIterator(identity_op, cupy_array, op_return_value_type)
```
This way we'd still have a test targeted at `RawPointer`, but through a public API.